level2.src.html

```diff
@@ -84,29 +84,90 @@ <h2 id="algorithms">Algorithms</h2>
     <h3 id="upgrade-algorithm">Upgrade |request| to an <i lang="la">a priori</i>
     authenticated URL as mixed content, if appropriate</h3>
 
-    TODO. This algorithm is called from Fetch before blocking requests as mixed content. It checks if the request is optionally-blockable mixed content, and if so rewrites the scheme to HTTPS.
+    Note: The Fetch specification will hook into this algorithm to upgrade optionally-blockable
+    mixed content to HTTPS.
+
+    Given a <a>Request</a> <var>request</var>, this algorithm will rewrite
+    its <a for="request">url</a> if the request is deemed to be optionally-blockable mixed content,
```

Comment:
What are your thoughts on upgrading blockable content as well? I know we talked about some firewalls blackholing unexpected TLS connections... it would be nice if we could just upgrade everything.

Reply:
I used to be in favor of autoupgrading blockable content, but I've been less enthusiastic lately. I'm worried that it will complicate shipping autoupgrading by causing more breakage. Also, the consequences of loading an unintended script are potentially a lot worse than an unintended image. (That is, if https://a.com/script.js and http://a.com/script.js are different resources, that seems possibly more problematic, e.g. by introducing a security vulnerability, than images.)

I think my preferred approach would be to get optionally-blockable autoupgrading out the door, and then consider if we want to do blockable too. WDYT?

Reply:
If I think upgrading blockable makes sense if the status quo is such that we don't error on blockable too often. I.e., it would primarily help sites transition. If the status quo is such that we already error a lot of blockable, then breakage or performance regressions seem more likely than a better experience.

Reply:
I'm happy to defer to your preference for this pass, @estark37. Seems like it'd be a good thing to discuss with other folks at TPAC?